tests/test_s3urlcache.py

```diff
@@ -0,0 +1,189 @@
+
+    @classmethod
+    def setUpClass(cls):
+        replica = Replica.aws
+        Config.set_config(BucketConfig.TEST_FIXTURE)
+        cls.blobstore, _, cls.test_fixture_bucket = Config.get_cloud_specific_handles(replica)
+        Config.set_config(BucketConfig.TEST)
+        cls.test_bucket = replica.bucket
+
+    def setUp(self):
+        self.urls_to_cleanup = []
+        self.cache = S3UrlCache(logger)
+
+    def tearDown(self):
+        self.delete_cached_urls(self.urls_to_cleanup)
+
+    def test_store_in_cache(self):
+        "The url contents are stored in s3 and the contents returned, when requested url is not found in cache."
+        url = f"{HTTPInfo.url}/{KB}"
+        self.urls_to_cleanup.append(url)
+        url_key = S3UrlCache._create_key(url)
+
+        self.delete_cached_urls(self.urls_to_cleanup)
+        with self.assertLogs(logger, "INFO") as log_monitor:
+            url_content = self.cache.resolve(url)
+
+        original_data = RANDOMDATA[:KB]
+        self.assertEqual(len(url_content), KB)
+        self.assertEqual(url_content, original_data)
+        self.assertTrue(log_monitor.output[0].endswith(f"{url} not found in cache. Added it to "
+                                                       f"{self.test_bucket} with key {url_key}."))
+
+    def test_retrieve_from_cache(self):
+        "Stored url contents is retrieved from s3, when a cached url is requested"
+        url = f"{HTTPInfo.url}/{KB}"
+        self.urls_to_cleanup.append(url)
+        with self.assertLogs(logger, 'INFO') as log_monitor:
+            url_content = self.cache.resolve(url)
+            cached_content = self.cache.resolve(url)
+        self.assertEqual(len(log_monitor.output), 1)
+        self.assertEqual(url_content, cached_content)
+
+    def test_bad_url(self):
+        bad_urls = ['', '//', 'http://?']
+        self.urls_to_cleanup.extend(bad_urls)
+
+        for url in bad_urls:
+            with self.subTest(bad_url=url):
+                with self.assertRaises(requests.RequestException):
+                    self.cache.resolve(url)
+
+    def test_url_SizeLimitError(self):
+        #  Exception returned when url content size is greater than max_size.
+        url = f"{HTTPInfo.url}/{KB}"
+        self.urls_to_cleanup.append(url)
+        self.cache.max_size = 1
+        with self.assertRaises(SizeLimitError) as ex:
+            self.cache.resolve(url)
+        self.assertEqual(ex.exception.args[0], f"{url} not cached. The URL's contents have exceeded "
+                                               f"{self.cache.max_size} bytes.")
+
+    def test_content_size_0(self):
+        url = f"{HTTPInfo.url}/0"
+        self.urls_to_cleanup.append(url)
+        url_content = self.cache.resolve(url)
+        self.assertEqual(len(url_content), 0)
+
+    def test_chunked_content(self):
+        size = KB * 10
+        url = f"{HTTPInfo.url}/{size}"
+        self.urls_to_cleanup.append(url)
+        self.cache.max_size = MB
```

Comment:
For readability, I would change this to `1 * MB` and `1 * KB`

Reply:
fixed